src/actions/repeated-action.h

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2011 - 2020 José Expósito <jose.exposito89@gmail.com>
+ *
+ * This file is part of Touchégg.
+ *
+ * Touchégg is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License  as  published by  the  Free Software
+ * Foundation,  either version 3 of the License,  or (at your option)  any later
+ * version.
+ *
+ * Touchégg is distributed in the hope that it will be useful,  but  WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+ * A PARTICULAR PURPOSE.  See the  GNU General Public License  for more details.
+ *
+ * You should have received a copy of the  GNU General Public License along with
+ * Touchégg. If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef ACTIONS_REPEATED_ACTION_H_
+#define ACTIONS_REPEATED_ACTION_H_
+
+#include "actions/action.h"
+
+class RepeatedAction : public Action {
+ public:
+  using Action::Action;
+  virtual ~RepeatedAction() = default;
+
+  /**
+   * Execute gesture on begin if threshold passed
+   */
+  void onGestureBegin(const Gesture &gesture) override;
+  /**
+   * Execute gesture repeatedly
+   */
+  void onGestureUpdate(const Gesture &gesture) override;
+  /**
+   * Execute gesture on end if threshold passed
+   */
+  void onGestureEnd(const Gesture &gesture) override;
+
+  /**
+   * If your action requires some pre-setup, override this method. This is
+   * executed unconditionally upon gesture beginning
+   */
+  virtual void executePrelude();
+  /**
+   * If your action requires some post-setup, override this method. This is
+   * executed unconditionally upon gesture ending
+   */
+  virtual void executePostlude();
```

Comment:
Should this be virtual pure?

```suggestion
  virtual void executePostlude() = 0;
```

Reply:
I don't think it should, I think prelude and postlude should be defaulted to noop for derived classes that don't need them, like `run-command` dosn't override `executePostlude` so nothing is executed there

Reply:
Fair enough, I prefer it virtual pure so it is clear what methods are children classes suppose to override plus it also makes clear that `RepeatedAction` is not mean to be instantiated. But it's up to you, I'm OK with both options, feel free to resolve this conversation if you want.